ext/web/06_streams.js

```diff
@@ -762,10 +763,67 @@
     }
   }
 
+  /**
+   * @param {ReadableStream<Uint8Array>} stream
+   * @returns {{ rid: number, autoClose: boolean }}
+   */
   function getReadableStreamResourceBacking(stream) {
     return stream[_resourceBacking];
   }
 
+  /**
+   * Create a new WritableStream object that is backed by a Resource that
+   * implements `Resource::write`. This object contains enough metadata to allow
+   * callers to bypass the JavaScript WritableStream implementation and write
+   * directly to the underlying resource if they so choose (FastStream).
+   * @param {number} rid The resource ID to write to.
+   * @param {boolean=} autoClose If the resource should be auto-closed when the stream closes. Defaults to true.
+   * @returns {WritableStream<Uint8Array>}
+   */
+  function writableStreamForRid(rid, autoClose = true) {
+    const stream = webidl.createBranded(WritableStream);
+    stream[_resourceBacking] = { rid, autoClose };
+    const underlyingSink = {
+      async write(chunk, controller) {
+        try {
+          let nwritten = 0;
+          while (nwritten < chunk.length) {
+            nwritten += await core.write(
+              rid,
+              TypedArrayPrototypeSubarray(chunk, nwritten),
+            );
+          }
+        } catch (e) {
+          controller.error(e);
+          if (autoClose) core.tryClose(rid);
+        }
+      },
+      close() {
+        if (autoClose) core.tryClose(rid);
+      },
+      abort() {
+        if (autoClose) core.tryClose(rid);
```

Comment:
Thought: maybe we should store a field to check if the stream has been closed so we don't have to call into Rust? Though it might be negligible cost